subscriber_notifee.go

```diff
@@ -96,10 +90,10 @@ func (nn *subscriberNotifee) subscribe(proc goprocess.Process) {
 				case dht.triggerSelfLookup <- nil:
 				default:
 				}
-			case event.EvtPeerIdentificationCompleted:
-				handlePeerIdentificationCompletedEvent(dht, evt)
 			case event.EvtPeerProtocolsUpdated:
 				handlePeerProtocolsUpdatedEvent(dht, evt)
+			case event.EvtPeerIdentificationCompleted:
+				dht.fixLowPeers()
```

Comment:
IIUC @petar  has been clear about the fact that connections != RT membership. Just because a peer connected to us/vice vera, does not mean that it's goo enough to be in the RT unless it's "useful" to us.

The only reason I've kept the `fixLowPeers` here is to "boostrap" the RT i.e. have some peers in it from which we can kick off queries/ensure it dosen't fall below a min threshold.

Reply:
This is done.

Reply:
Where did you add the documentation? Also, this trips up some of our own go tests as well. As for connections != RT membership, we wouldn't add them to the routing table we're just proposing them.

If they happen to kick out a useless peer then so be it, but there's not really a reason to treat them any worse then peers that connect to us and then send us a query. While this could potentially bug us in the future if we kick peers out of the routing table probabilistically (since we'll be giving peers effectively two shots at getting into the table) it seems intuitive (at least to me) that if A manually connects to B that they will be added to each others routing tables if there's space available. This is especially true since we have no publicly exposed APIs that allow "send a DHT query/ping to peer B". On the other hand, this is most important when the number of peers is low and `fixLowPeers` should save us here.

lmk what you think.

Reply:
@aschmahmann 

1. It could be a peer that supports the DHT making a connection for Gossipsub. There's no value add there.
2. We now add inbound/outbound peers with a `lastOutBoundQueryTime=time.Now` to prevent them from being evicted before one "round". This means we could end up blocking the addition of peers who actually make a query to us/get one from us just because we are connected to DHT peers who might actually be of no use.

I've added the docs at the top of `dht.New`.

Reply:
After an offline @aarshkshah1992  and I decided that we would add the peers to the routing table, but wouldn't give them a starting `lastOutboundQueryTime` which would make them instantly replaceable by other peers that the node encounters or queries if the bucket was full.